src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.h

```diff
@@ -0,0 +1,688 @@
+    uint8_t mCENCKeyIDBuffer[kMaxCENCKeyIDLength];
+};
+
+struct ContainerOptionsStorage : public ContainerOptionsStruct
+{
+    ContainerOptionsStorage(){};
+
+    ContainerOptionsStorage(const ContainerOptionsStorage & aContainerOptionsStorage) { *this = aContainerOptionsStorage; }
+
+    ContainerOptionsStorage & operator=(const ContainerOptionsStorage & aContainerOptionsStorage)
+    {
+        containerType = aContainerOptionsStorage.containerType;
+
+        mCMAFContainerStorage = aContainerOptionsStorage.mCMAFContainerStorage;
+
+        CMAFContainerOptions.SetValue(mCMAFContainerStorage);
+
+        return *this;
+    }
+
+    ContainerOptionsStorage(Structs::ContainerOptionsStruct::DecodableType containerOptions)
+    {
+        containerType = containerOptions.containerType;
+
+        if (containerType == ContainerFormatEnum::kCmaf)
+        {
+            mCMAFContainerStorage = CMAFContainerOptionsStorage(containerOptions.CMAFContainerOptions);
+            CMAFContainerOptions.SetValue(mCMAFContainerStorage);
+        }
+        else
+        {
+            CMAFContainerOptions.Missing();
+        }
+    }
+
+private:
+    CMAFContainerOptionsStorage mCMAFContainerStorage;
+};
+
+struct TransportOptionsStorage : public TransportOptionsStruct
+{
+    TransportOptionsStorage(){};
+
+    TransportOptionsStorage(const TransportOptionsStorage & aTransportOptionsStorage) { *this = aTransportOptionsStorage; }
+
+    TransportOptionsStorage & operator=(const TransportOptionsStorage & aTransportOptionsStorage)
+    {
+        streamUsage   = aTransportOptionsStorage.streamUsage;
+        videoStreamID = aTransportOptionsStorage.videoStreamID;
+        audioStreamID = aTransportOptionsStorage.audioStreamID;
+        endpointID    = aTransportOptionsStorage.endpointID;
+
+        // Deep copy the URL buffer
+        std::memcpy(mUrlBuffer, aTransportOptionsStorage.mUrlBuffer, kMaxUrlLength);
+        url = MutableCharSpan(mUrlBuffer, aTransportOptionsStorage.url.size());
+
+        // Copy internal storage objects
+        mTriggerOptionsStorage = aTransportOptionsStorage.mTriggerOptionsStorage;
+        triggerOptions         = mTriggerOptionsStorage;
+
+        ingestMethod = aTransportOptionsStorage.ingestMethod;
+
+        mContainerOptionsStorage = aTransportOptionsStorage.mContainerOptionsStorage;
+        containerOptions         = mContainerOptionsStorage;
+
+        expiryTime = aTransportOptionsStorage.expiryTime;
+
+        return *this;
+    }
+
+    TransportOptionsStorage(Structs::TransportOptionsStruct::DecodableType transportOptions)
+    {
+        streamUsage   = transportOptions.streamUsage;
+        videoStreamID = transportOptions.videoStreamID;
+        audioStreamID = transportOptions.audioStreamID;
+        endpointID    = transportOptions.endpointID;
+
+        MutableCharSpan urlBuffer(mUrlBuffer);
+        CopyCharSpanToMutableCharSpanWithTruncation(transportOptions.url, urlBuffer);
```

Comment:
Again, is something upstream of here validating the length?  This is safe, because WithTruncation, but ideally too-long things would get errors before this point.